.github/workflows/run-tests.yml

```
@@ -2,15 +2,13 @@ name: Run Tests
 
 on: pull_request
 
-env:
-  ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
-
 jobs:
   run_tests:
     name: Run Tests
-    runs-on: ubuntu-latest
+    # Becomes ubuntu-latest soon, required to run node20 actions.
+    runs-on: ubuntu-24.04
     container:
-      image: perl:5.28.1
+      image: perl:5.40.0
     steps:
       - uses: actions/checkout@v3
```

I think it's probably safe to move to upgrade this, too.
 
```suggestion
      - uses: actions/checkout@v4
```